src/main/java/teammates/logic/api/GateKeeper.java

```diff
@@ -324,24 +324,34 @@ private boolean isStudent() {
     }
 
     public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttributes account, String courseId,
-            String section, String email) {
+                                                                       String section, String email) {
+        if (!isValidInstructor(section, courseId, account) && !isValidStudent(email, courseId, account)) {
+            throw new UnauthorizedAccessException("User is not in the course that student belongs to");
+        }
+    }
+
+    private boolean isValidInstructor(String section, String courseId, AccountAttributes account) {
         InstructorAttributes instructor = instructorsLogic.getInstructorForGoogleId(courseId, account.googleId);
-        if (instructor != null) {
-            if (!instructor.isAllowedForPrivilege(section,
-                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
-                throw new UnauthorizedAccessException("Instructor does not have enough privileges to view the photo");
-            }
-            return;
+        if (instructor == null) {
+            return false;
         }
 
+        if (!instructor.isAllowedForPrivilege(section,
+                                              Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
+            throw new UnauthorizedAccessException("Instructor does not have enough privileges to view the photo");
```

Comment:
Agree with the concept that boolean methods should not be throwing exceptions. However, this approach would cause errors due to null instructors becoming indistinguishable from errors due to insufficient privileges. I suggest that the null checks can remain in the original method instead:
```java
public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttributes account, String courseId,
        String section, String email) {
    InstructorAttributes instructor = instructorsLogic.getInstructorForGoogleId(courseId, account.googleId);
    if (instructor != null) {
        verifyInstructorCanViewPhoto(...);
        return;
    }

    StudentAttributes student = studentsLogic.getStudentForCourseIdAndGoogleId(courseId, account.googleId);
    if (student != null) {
        verifyStudentCanViewPhoto(...);
        return;
    }

    throw new UnauthorizedAccessException("User is not in the course that student belongs to");
}
```

@Zhiyuan-Amos would this be concise enough?